### DISCUSSION TITLE
API Preview crashing for tables in custom schema

### DISCUSSION THREAD

**Discussion by saltcod** [supabase/supabase#16841] [Contribute to Supabase]
# Bug report

## Describe the bug

The API preview in the table editor crashes when viewing a table in a custom schema

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a new schema
2. Create a new table in that custom schema
3. Select that table from the Table Editor 
4. Click the API button  
5. Page crashes

## Expected behavior

You should be able to view the API docs for a table in a custom schema
![CleanShot 2023-06-02 at 16 25 33@2x](https://github.com/supabase/supabase/assets/105593/ef4e89d6-1231-4e9d-bea2-574daf19d60a)
<img width="1391" alt="CleanShot 2023-06-02 at 16 30 10@2x" src="https://github.com/supabase/supabase/assets/105593/c2be07c3-73b3-4faa-abb6-2156334f850c">

**Comment by mqshaikh8** [comment]
Is this still an issue? I am able to do all of this without it crashing. After I clicked the api button, a drawer opens up from the right.
<img width="1469" alt="Screen Shot 2023-06-22 at 5 29 21 PM" src="https://github.com/supabase/supabase/assets/46161085/a861edbd-fa6e-46b3-89b8-04d35f2e4cc4">

**Comment by soedirgo** [comment]
This happens because the <kbd>API</kbd> button relies on PostgREST's OpenAPI output, which only contains schemas included in the <kbd>Exposed schemas</kbd> setting in https://supabase.com/dashboard/project/_/settings/api .

These exposed schemas are retrieved from https://api.supabase.io/platform/projects/_/config/postgrest . The fix should involve hiding the <kbd>API</kbd> button if the table's schema is not `db_schema`.

**Reply by aviroopjana** [reply]
@soedirgo hey! I want to fix this issue can you help me fix it... what should I do to fix it, I am very new here
